src/asmcup/genetics/Gene.java

```diff
@@ -10,16 +12,27 @@ public Gene(byte[] dna, int gen, float score) {
 		this.gen = gen;
 		this.score = score;
 	}
-	
+
+	@Override
 	public int compareTo(Gene other) {
 		float d = score - other.score;
 		
-		if (d == 0) {
+		if (Math.abs(d) < 0.00000001f) {

```

Comment:
(Note: I'm not arguing because I'm trying to "fight" over this, just trying to make the intention of the code clear.)

> Let's say you will increase score of Gene A by 0.7f and score of Gene B by 0.37f and then 0.33f.

This cannot happen because the scores are final. But even if it were possible:

> I understand that you are creating an ordering. But it's arbitrary.

...not more arbitrary than the float calculations. All we're trying to do is order the genes by score. If, due to whatever calculations happened before, two scores that should be equal are not equal, then so be it. For what it's worth to the comparison of genes, one of the two is bigger, which should be reflected.

In other words, the goal of the code in `compareTo()` is not primarily to determine equality. In fact, in an initial version there was no check for equality, so you could end up with `!(a > b)` and `!(b > a)` (and by consequence, `a > b` and `b > a`) both being true, which every sorting algorithm will (and did) throw a fit over. That's why a check for (binary) equality was introduced, just to weed out that single edge case.